### ISSUE TITLE
User settings not saved after app restart

### ISSUE THREAD

**Issue by Yann-La-Baguette** [soleon/Percentage#4]
At each restart of my PC, the application resets all the settings I setted.

**Comment by Yann-La-Baguette** [comment]
Yup, I have .NET 9.0.101 to be precise. It does'nt work either :/
It's not a big problem, so it doesn't matter if it's not solved quickly. It was mainly to let you know that this bug exists

**Comment by Yann-La-Baguette** [comment]
I thought you were, nevermind

**Comment by soleon** [comment]
Both the released portable exe file and the Microsoft Store release are self-contained packages. It contains all the necessary files from .NET 9.0 to run the app so no separate .NET 9.0 runtime is needed.

Not saving the notification toggle settings was a bug fixed in [release 2.0.8](https://github.com/soleon/Percentage/releases/tag/2.0.8), the previous release [2.0.7](https://github.com/soleon/Percentage/releases/tag/2.0.7) also has an update to immediately save user settings in case Windows force shutdown the app during restart or the app crashes losing all the unsaved changes. Although I can't reproduce losing my settings even if I restart with versions prior to 2.0.7.

The app saves user settings in
```
C:\Users\<user name>\AppData\Local\Percentage.App\Percentage.App_Url_<unique ID>\1.0.0.0\user.config
```
The `unique ID` part is different between compiling and running from source code, the released app exe and the Microsoft Store version. As long as you keep running the app from the same source, the path should always be the same. The version number in the path is always going to be `1.0.0.0` to avoid losing user settings when you update the app.

If the app can't access this file due to your security config, it won't be able to save or read any user settings.

Here is an example `user.config` that has everything customised.
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <userSettings>
        <Percentage.App.Properties.Settings>
            <setting name="HideAtStartup" serializeAs="String">
                <value>True</value>
            </setting>
            <setting name="TrayIconFontFamily" serializeAs="String">
                <value>Microsoft YaHei UI</value>
            </setting>
            <setting name="RefreshSeconds" serializeAs="String">
                <value>1</value>
            </setting>
            <setting name="TrayIconFontBold" serializeAs="String">
                <value>True</value>
            </setting>
            <setting name="TrayIconFontUnderline" serializeAs="String">
                <value>True</value>
            </setting>
            <setting name="BatteryCriticalColour" serializeAs="String">
                <value>#EA005E</value>
            </setting>
            <setting name="BatteryLowColour" serializeAs="String">
                <value>#F7630C</value>
            </setting>
            <setting name="BatteryChargingColour" serializeAs="String">
                <value>#00CC6A</value>
            </setting>
            <setting name="BatteryNormalColour" serializeAs="String">
                <value>#00B7C3</value>
            </setting>
            <setting name="BatteryHighNotification" serializeAs="String">
                <value>False</value>
            </setting>
            <setting name="BatteryFullNotification" serializeAs="String">
                <value>False</value>
            </setting>
            <setting name="BatteryCriticalNotification" serializeAs="String">
                <value>False</value>
            </setting>
            <setting name="BatteryLowNotification" serializeAs="String">
                <value>False</value>
            </setting>
            <setting name="BatteryHighNotificationValue" serializeAs="String">
                <value>75</value>
            </setting>
            <setting name="BatteryLowNotificationValue" serializeAs="String">
                <value>25</value>
            </setting>
            <setting name="BatteryCriticalNotificationValue" serializeAs="String">
                <value>15</value>
            </setting>
        </Percentage.App.Properties.Settings>
    </userSettings>
</configuration>
```

**Comment by Yann-La-Baguette** [comment]
I reinstalled it entirely, it works well now. Maybe one of my settings was blocking the application from accessing the user.config file like you telled me.
I think your update made it works again, thanks.
